`cross bench` fails to link with "error adding symbols: File in wrong format"

Steps:

- create a new empty project `cargo new cross-test`
- run `cross bench --target=...` with ~any target (I tried `powerpc-unknown-linux-gnu`, `mips64-unknown-linux-gnuabi64`, `aarch64-unknown-linux-gnu`)

Output:

```
cross bench --target=powerpc-unknown-linux-gnu
   Compiling cross-test v0.1.0 (/home/vagrant/cross-test)                                                                                  
error: linking with `cc` failed: exit code: 1                                                                                              
  |                                                                                                                                        
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m32" "-L" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib" "/home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o" "/home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.1.rcgu.o" "-o" "/home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e" "/home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.lwh4u3jo28fe8ll.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps" "-L" "/home/vagrant/cross-test/target/release/deps" "-L" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libtest-c0479f7cd36f68a3.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libterm-aaa34b751178d353.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libgetopts-f72f6430dcc20e65.rlib" "-Wl,--start-group" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libstd-f1140b6fc9418a82.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libpanic_unwind-b89f1a9b548bfef9.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libunwind-4c9cbe622de116ab.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/liballoc_system-48c8a8552cff934e.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/liblibc-517830e9d56bab97.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/liballoc-2d44723f32308cf8.rlib" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libcore-f2133e8b70369157.rlib" "-Wl,--end-group" "/home/vagrant/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/powerpc-unknown-linux-gnu/lib/libcompiler_builtins-71671e8958739d51.rlib" "-Wl,-Bdynamic" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /usr/bin/ld: cannot find Scrt1.o: No such file or directory                                                                      
          /usr/bin/ld: cannot find crti.o: No such file or directory                                                                       
          /usr/bin/ld: /home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o: Relocations in generic ELF (EM: 20)
          /usr/bin/ld: /home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o: Relocations in generic ELF (EM: 20)
          /usr/bin/ld: /home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o: Relocations in generic ELF (EM: 20)
          /usr/bin/ld: /home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o: Relocations in generic ELF (EM: 20)
          /home/vagrant/cross-test/target/powerpc-unknown-linux-gnu/release/deps/cross_test-de7cb8390fbe8a1e.cross_test.yt4erta5-cgu.0.rcgu.o: error adding symbols: File in wrong format
          collect2: error: ld returned 1 exit status
```

`cross test --release --target=...` *does* work.

--- a/ci/script.sh
+++ b/ci/script.sh
@@ -91,6 +91,7 @@ EOF
             pushd $td
             cargo init --lib --name foo .
             cross_test --target $TARGET
+            cross_bench --target $TARGET
             popd
 
             rm -rf $td
@@ -124,6 +125,7 @@ EOF
                 cross_run --target $TARGET
                 cross_run --target $TARGET --example e
                 cross_test --target $TARGET
+                cross_bench --target $TARGET
                 popd
 
                 rm -rf $td
@@ -192,6 +194,17 @@ cross_test() {
         for runner in $RUNNERS; do
             echo -e "[target.$TARGET]\nrunner = \"$runner\"" > Cross.toml
             cross test "$@"
+        done
+    fi
+}
+
+cross_bench() {
+    if [ -z "$RUNNERS" ]; then
+        cross bench "$@"
+    else
+        for runner in $RUNNERS; do
+            echo -e "[target.$TARGET]\nrunner = \"$runner\"" > Cross.toml
+            cross bench "$@"
         done
     fi
 }

--- a/src/cargo.rs
+++ b/src/cargo.rs
@@ -13,6 +13,7 @@ pub enum Subcommand {
     Run,
     Rustc,
     Test,
+    Bench,
     Deb,
 }
 
@@ -26,7 +27,7 @@ impl Subcommand {
 
     pub fn needs_interpreter(&self) -> bool {
         match *self {
-            Subcommand::Run | Subcommand::Test => true,
+            Subcommand::Run | Subcommand::Test | Subcommand::Bench => true,
             _ => false,
         }
     }
@@ -40,6 +41,7 @@ impl<'a> From<&'a str> for Subcommand {
             "run" => Subcommand::Run,
             "rustc" => Subcommand::Rustc,
             "test" => Subcommand::Test,
+            "bench" => Subcommand::Bench,
             "deb" => Subcommand::Deb,
             _ => Subcommand::Other,
         }

